sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/AccessPolicyAsyncTests.java

```diff
@@ -0,0 +1,1024 @@
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@Isolated
+public class AccessPolicyAsyncTests extends BlobTestBase {
+    private BlockBlobAsyncClient blockBlobAsyncClient;
+    private AppendBlobAsyncClient appendBlobAsyncClient;
+    private PageBlobAsyncClient pageBlobAsyncClient;
+    private final String scope1 = "testscope1";
+    private BlockBlobAsyncClient cpknBlockBlob;
+    private BlobContainerAsyncClient blobContainerClient;
+    private BlobAsyncClient blobVersionedClient;
+
+    @BeforeEach
+    public void setup() {
+        blockBlobAsyncClient = ccAsync.getBlobAsyncClient(generateBlobName()).getBlockBlobAsyncClient();
+        blockBlobAsyncClient.upload(DATA.getDefaultFlux(), DATA.getDefaultDataSize(), true).block();
+
+        appendBlobAsyncClient = ccAsync.getBlobAsyncClient(generateBlobName()).getAppendBlobAsyncClient();
+        appendBlobAsyncClient.create().block();
+
+        pageBlobAsyncClient = ccAsync.getBlobAsyncClient(generateBlobName()).getPageBlobAsyncClient();
+        pageBlobAsyncClient.create(PageBlobClient.PAGE_BYTES).block();
+
+        BlobContainerClientBuilder builder = getContainerClientBuilder(ccAsync.getBlobContainerUrl())
+            .credential(ENVIRONMENT.getPrimaryAccount().getCredential());
+        BlobContainerAsyncClient cpknContainer = builder.encryptionScope(scope1).buildAsyncClient();
+        cpknBlockBlob = cpknContainer.getBlobAsyncClient(generateBlobName()).getBlockBlobAsyncClient();
+
+        blobContainerClient = versionedBlobServiceAsyncClient.createBlobContainer(generateContainerName()).block();
+        blobVersionedClient = blobContainerClient.getBlobAsyncClient(generateBlobName());
+    }
+
+    //Container Async Tests
+
+    @Test
+    public void setAccessPolicyMinAccess() {
+        setAccessPolicySleepAsync(ccAsync, PublicAccessType.CONTAINER, null);
+        StepVerifier.create(ccAsync.getProperties())
+            .assertNext(r -> assertEquals(r.getBlobPublicAccess(), PublicAccessType.CONTAINER))
+            .verifyComplete();
+    }
+
+    @Test
+    public void setAccessPolicyMinIds() {
+        BlobSignedIdentifier identifier = new BlobSignedIdentifier()
+            .setId("0000")
+            .setAccessPolicy(new BlobAccessPolicy()
+                .setStartsOn(testResourceNamer.now().atZoneSameInstant(ZoneId.of("UTC")).toOffsetDateTime())
+                .setExpiresOn(testResourceNamer.now().atZoneSameInstant(ZoneId.of("UTC")).toOffsetDateTime()
+                    .plusDays(1))
+                .setPermissions("r"));
+
+        List<BlobSignedIdentifier> ids = Collections.singletonList(identifier);
+
+        setAccessPolicySleepAsync(ccAsync, null, ids);
+
+        StepVerifier.create(ccAsync.getAccessPolicy())
+            .assertNext(r -> assertEquals(r.getIdentifiers().get(0).getId(), "0000"))
```

Comment:
_This may apply to more locations than just this_

Flip the values in the `assertEquals`, the first value is the expected value and the second value is the actual value. This doesn't affect correctness of the test but it may make debugging more confusing